### PR TITLE
CXF-83184 Adding Virtual Device & Adding VD 2 Azure Example

### PR DESCRIPTION
**Pull request by srushti-patl**
Includes:
 Virtual Device Connection Module
VD 2 Azure Connection example using NE module

### CODE REVIEW

File: examples/virtual-device-2-azure-connection/main.tf [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+
+  connection_name       = var.connection_name
+  connection_type       = var.connection_type
+  notifications_type    = var.notifications_type
+  notifications_emails  = var.notifications_emails
+  bandwidth             = azurerm_express_route_circuit.port2azure.bandwidth_in_mbps
+  purchase_order_number = var.purchase_order_number
+
+  # A-side
+  aside_ap_type = var.aside_ap_type
```
Comment (thogarty) [inline examples/virtual-device-2-azure-connection/main.tf:88]: Are these variables necessary? Are we making the users choose AP type and other types in the other modules?

Comment (thogarty) [inline examples/virtual-device-2-azure-connection/main.tf:88]: Are these variables necessary? Are we making the users choose AP type and other types in the other modules?

File: examples/virtual-device-2-azure-connection/main.tf [hunk 2]
```diff
@@ -0,0 +1,99 @@
+provider "equinix" {
+  client_id     = var.equinix_client_id
+  client_secret = var.equinix_client_secret
+  endpoint      = "https://uatapi.equinix.com"
```
Comment (thogarty) [inline examples/virtual-device-2-azure-connection/main.tf:4]: Shouldn't leave testing endpoints in examples that customers can use.

**Review by thogarty — CHANGES_REQUESTED** [review]
Changes requested. See comments.

File: examples/virtual-device-2-azure-connection/main.tf (hunk 1 above)
Comment (srushti-patl) [reply examples/virtual-device-2-azure-connection/main.tf:88]: No, we are not making users choose AP for aside. I will make the changes.

File: modules/virtual-device-connection/main.tf [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+
+  a_side {
+    access_point {
+      type = var.aside_ap_type
+      virtual_device {
+        type = var.aside_vd_type
+        uuid = var.aside_vd_uuid
+      }
+      interface {
+        type = var.aside_interface_type != "" ? var.aside_interface_type : null
```
[the remark this answers is not in the document]
Comment (srushti-patl) [reply modules/virtual-device-connection/main.tf:37]: There is only one type of virtual device i.e EDGE.

**Review by thogarty — APPROVED** [review]
LGTM! Shipping it.

**Pull request merged**